source/gui/startupDialogs.py

```diff
@@ -104,7 +105,26 @@ def __init__(self, parent):
 	def onOk(self, evt):
 		layout = self.kbdNames[self.kbdList.GetSelection()]
 		config.conf["keyboard"]["keyboardLayout"] = layout
-		config.conf["keyboard"]["useCapsLockAsNVDAModifierKey"] = self.capsAsNVDAModifierCheckBox.IsChecked()
+		NVDAKeysVal = (
+			(NVDAKey.CAPS_LOCK.value if self.capsAsNVDAModifierCheckBox.IsChecked() else 0)
+			| (config.conf["keyboard"]["NVDAModifierKeys"] & NVDAKey.NUMPAD_INSERT.value)
+			| (config.conf["keyboard"]["NVDAModifierKeys"] & NVDAKey.EXTENDED_INSERT.value)
+		)
+		if NVDAKeysVal == 0:
+			log.debugWarning("No NVDA key set")
+			gui.messageBox(
+				_(
+					# Translators: The title of an error message box displayed when validating the startup dialog
+					"Despite your choice, caps lock will remain used as NVDA modifier key"
+					" since no other possible key is currently configured as NVDA modifier key."
+				),
+				# Translators: The title of an error message box displayed when validating the startup dialog
+				_("Error"),
+				wx.OK | wx.ICON_ERROR,
+				self,
+			)
```

Comment:
I think this should be explicitly set here, in case the user had a different initial config.
Otherwise the message should indicate that the NVDA modifier key will not change, rather than be set to caps lock.

```suggestion
			)
			config.conf["keyboard"]["NVDAModifierKeys"] = NVDAKeysVal.CAPS_LOCK
```

Reply:
When there is an issue with the checkbox value, it's better not to save anything new in the config regarding NVDA key. In 61af5f252be7acb40e51cb7aef5f0992a90b3e79, the messagenow  indicates that nothing will change.